File: frontend/src/types/content.ts
export type ContentType = "video" | "document" | "article" | "image";

export type SavedContentItem = {
  id?: string;
  slug: string;
  title: string;
  type: ContentType;
  source: string;
  sourceUrl?: string;
  thumbnailUrl?: string;
  description: string;
  metadata: string;
  dateLabel: string;
  spaceSlug: string;
  spaceName: string;
  tags: string[];
  status?: string;
  creator?: string;
  duration?: string;
  platform?: "youtube" | "instagram" | "facebook" | string;
  detail: {
    heroLabel: string;
    previewTitle: string;
    previewBody: string;
    extractedTitle: string;
    extractedBody: string[];
    referenceLabel: string;
    references: string[];
  };
};

export type Space = {
  slug: string;
  name: string;
  description: string;
  ownerLabel: string;
  updatedAt: string;
  tags: string[];
};

File: backend/internal/models/content.go
package models

import "time"

type ContentType string

const (
	ContentTypeVideo    ContentType = "video"
	ContentTypeDocument ContentType = "document"
	ContentTypeArticle  ContentType = "article"
	ContentTypeImage    ContentType = "image"
)

type Content struct {
	ID           string      `json:"id"`
	UserID       string      `json:"user_id"`
	SpaceID      string      `json:"space_id"`
	Title        string      `json:"title"`
	Description  string      `json:"description"`
	Type         ContentType `json:"type"`
	SourceURL    *string     `json:"source_url,omitempty"`
	ThumbnailURL *string     `json:"thumbnail_url,omitempty"`
	CreatedAt    time.Time   `json:"created_at"`
	UpdatedAt    time.Time   `json:"updated_at"`
}
